modules/services/clipse.nix

```diff
@@ -0,0 +1,109 @@
+{ pkgs, config, lib, ... }:
+let
+  cfg = config.services.clipse;
+  jsonFormat = pkgs.formats.json { };
+
+in with lib;
+
+{
+  meta.maintainers = [ lib.maintainers.dsoverlord ];
+
+  options.services.clipse = {
+    enable = mkEnableOption "Enable clipse clipboard manager";
+
+    package = mkPackageOption pkgs "clipse" { };
+
+    systemdTarget = lib.mkOption {
+      type = lib.types.str;
+      default = "graphical-session.target";
+      example = "sway-session.target";
+      description = ''
+        The systemd target that will automatically start the clipse service.
+
+        When setting this value to `"sway-session.target"`,
+        make sure to also enable {option}`wayland.windowManager.sway.systemd.enable`,
+        otherwise the service may never be started.
+      '';
+    };
+
+    allowDuplicates = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Allow duplicates";
+    };
+
+    historySize = mkOption {
+      type = types.int;
+      default = 100;
+      description = "Number of history lines to keep.";
+    };
+
+    theme = mkOption {
+      type = jsonFormat.type;
+
+      default = { useCustomTheme = false; };
+
+      example = literalExpression ''
+        {
+          useCustomTheme = true;
+          DimmedDesc = "#ffffff";
+          DimmedTitle = "#ffffff";
+          FilteredMatch = "#ffffff";
+          NormalDesc = "#ffffff";
+          NormalTitle = "#ffffff";
+          SelectedDesc = "#ffffff";
+          SelectedTitle = "#ffffff";
+          SelectedBorder = "#ffffff";
+          SelectedDescBorder = "#ffffff";
+          TitleFore = "#ffffff";
+          Titleback = "#434C5E";
+          StatusMsg = "#ffffff";
+          PinIndicatorColor = "#ff0000";
+        };
+      '';
+
+      description = ''
+        Configuration written to
+        {file}`$XDG_CONFIG_HOME/clipse/custom_theme.json`.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.clipse" pkgs
+        lib.platforms.linux)
```

Comment:
Looks like upstream supports more platforms, but we can leave the assertion until someone wants to add support in the module for others.